src/app/models/User.ts

```diff
@@ -175,23 +204,15 @@ export class SessionUser extends User implements SessionUserInterface {
     super(sessionUser);
   }
 
-  static fromJSON = (obj: any) => {
-    if (typeof obj === "string") {
-      obj = JSON.parse(obj);
-    }
-
-    return new SessionUser(obj);
-  };
-
-  toJSON() {
+  public toJSON() {
     return {
       authToken: this.authToken,
```

Comment:
not writeable

Reply:
This is required currently because of the `handleAuth` function in the security service. When a user logs in, i get a User model which i then convert into a Session User model: https://github.com/QutEcoacoustics/workbench-client/blob/master/src/app/services/baw-api/security.service.ts#L84-L90. Potentially there is a better was of doing this, however my assumption was that the server would be able to take only the values it needed.

Reply:
The only model that should have the auth token is the current user model. Even then,l it doesn't need it - the auth token shouldn't be part of any User model.

Thus:
- Ensure the current session/user is passed to the security service (injected, not as part of anyhting that will get toJSON'ed)
- access the the `user.authToken` as you would normally.

Reply:
- The `authToken` is not part of the `User` model. It is only part of the `SessionUser` model.
- I'm not sure what you mean by injecting the current `SessionUser` model into the security service, as the service is what it generating the model
- Removing the functionality from the `toJSON()` in the `User` model would still not work because I need to store the model inside localstorage and `JSON.stringify` will use the output of `toJSON()`

Reply:
Ah local storage serialization. Fair enough, carry on